src/main/java/com/spotify/dns/SimpleLookupFactory.java

```diff
@@ -21,20 +21,43 @@
 import org.xbill.DNS.Resolver;
 import org.xbill.DNS.TextParseException;
 import org.xbill.DNS.Type;
+import org.xbill.DNS.lookup.LookupSession;
 
-/**
- * A LookupFactory that always returns new instances.
- */
-public class SimpleLookupFactory implements LookupFactory {
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ForkJoinPool;
 
+/** A LookupFactory that always returns new instances. */
+public class SimpleLookupFactory implements LookupFactory {
+  private final LookupSession session;
   private final Resolver resolver;
 
+  /**
+   * @deprecated
+   * Deprecated to avoid overloading forkjoin common pool.
+   * Use {@link SimpleLookupFactory#SimpleLookupFactory(Executor)} instead.
+   */
+  @Deprecated
   public SimpleLookupFactory() {
-    this(null);
+    this(Lookup.getDefaultResolver());
   }
 
+  /**
+   * @deprecated
+   * Deprecated to avoid overloading forkjoin common pool.
+   * Use {@link SimpleLookupFactory#SimpleLookupFactory(Resolver, Executor)} instead.
+   */
   public SimpleLookupFactory(Resolver resolver) {
+    this(resolver, ForkJoinPool.commonPool());
+  }
+
+  public SimpleLookupFactory(Executor executor) {
+    this(Lookup.getDefaultResolver(), executor);
+  }
+
+  public SimpleLookupFactory(Resolver resolver, Executor executor) {
```

Comment:
Consider accepting `null` for `executor` and default to `ForkJoinPool.commonPool()` (or, if you don't like the pattern, ensure `executor` is not null).